### PR TITLE
remove errCh usage from HealObjects() simplify it

### LINKED ISSUE #14247: restarted minio process show zero disks
Hi, 

We've restart ( not a crash)  a minio service on one cluster node, and after 30 minutes we still have :
```
 ● REDACTED1:9000
   Uptime: 30 minutes
   Version: 2022-02-01T18:00:14Z
   Network: 4/4 OK
   Drives: 0/0 OK

●  REDACTED2:9000
   Uptime: 20 hours
   Version: 2022-02-01T18:00:14Z
   Network: 4/4 OK
   Drives: 4/4 OK

●  REDACTED3:9000
   Uptime: 20 hours
   Version: 2022-02-01T18:00:14Z
   Network: 4/4 OK
   Drives: 4/4 OK

●  REDACTED4:9000
   Uptime: 20 hours
   Version: 2022-02-01T18:00:14Z
   Network: 4/4 OK
   Drives: 4/4 OK
```

The node is up, but does not show the 4 configured drives.
There is no information in the console.

When we try to restart the node there is this line printed in the console :
`ERROR Server startup canceled upon user request: Initializing sub-systems stopped gracefully context canceled
`
When we inspect the process using lsof/strace, we see some disk i/o activity for a few seconds, reading the metadata of all buckets, then only network i/o with other nodes, no file stat or open.

## Expected Behavior

A functional minio node

## Current Behavior

The process is running, the node is up, respond to health checks, but no longer expose disks

## Steps to Reproduce (for bugs)

1. kill a minio node part of a cluster with a term signal

## Your Environment

* Version used (`minio --version`):` minio version RELEASE.2022-02-01T18-00-14Z`
* Server setup and configuration: `minio server https://minio{1...4}.REDACTED.com:9000/mnt/disk{1...4}/data --console-address ":9001"`

* Operating System and version (`uname -a`): `Linux REDACTED 5.10.0-3-amd64 #1 SMP Debian 5.10.13-1 (2021-02-06) x86_64 GNU/Linux
`

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
remove errCh usage from HealObjects() simplify it

## Motivation and Context
errCh is not needed instead, rely on errs slice to
capture and return errors instead.

most probably fixes #14247

## How to test this PR?
Fixes "rolling restart" hangs on HealObjects()

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by klauspost — APPROVED** [review]

File: cmd/erasure-server-pool.go [hunk 1]
```diff
@@ -1829,34 +1829,33 @@ func (z *erasureServerPools) HealObjects(ctx context.Context, bucket, prefix str
...
-			err = e
-			cancel()
+		poolErrs = append(poolErrs, errs)
+	}
+	for _, errs := range poolErrs {
+		for _, err := range errs {
+			if err == nil {
+				continue
+			}
+			return err
```
Comment (vadmeste) [inline cmd/erasure-server-pool.go:1855]:
```
if err != nil {
   return err
}
```

**Review by vadmeste — COMMENTED** [review]
Nit comment, LGTM otherwise

File: cmd/erasure-server-pool.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/erasure-server-pool.go:1855]: I didn't do that to avoid too much nesting, to allow a slight focus on err

Comment (harshavardhana) [reply cmd/erasure-server-pool.go:1855]: @vadmeste ^^

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**